hwbench/engines/fio.py

```diff
@@ -0,0 +1,213 @@
+            )
+
+    def need_skip_because_version(self):
+        if self.skip:
+            # we already skipped this benchmark, we can't know the reason anymore
+            # because we might not have run the version command.
+            return ["echo", "skipped benchmark"]
+        if not self.version_compatible():
+            print(f"WARNING: skipping benchmark {self.name}, needs fio >= 3.19")
+            self.skip = True
+            return ["echo", "skipped benchmark"]
+        return None
+
+    def run_cmd(self) -> list[str]:
+        skip = self.need_skip_because_version()
+        if skip:
+            return skip
+
+        # Let's build the command line to run the tool
+        args = [
+            self.engine_module.get_engine().get_binary(),
+        ]
+
+        return self.get_taskset(args)
+
+    def get_default_fio_command_line(self, args: list) -> list:
+        """Return the default fio arguments"""
+
+        def remove_arg(args, item) -> list:
+            if isinstance(item, str):
+                return [arg for arg in args if not arg.startswith(item)]
+            else:
+                # We need to ensure that value based items are having the right value
+                # This avoid a case where the user already defined a value we need to control
+                for arg in args:
+                    if arg.startswith(item[0]):
+                        if arg != f"{item[0]}={item[1]}":
+                            print(
+                                f"{self.parameters.get_name_with_position()}: Fio parameter {item[0]} is now set to {item[1]}"
+                            )
+                            args.remove(arg)
+
+                return args
+
+        name = self.parameters.get_name_with_position()
+        enforced_items = [
+            ["--runtime", f"{self.parameters.get_runtime()}"],
+            "--time_based",
+            ["--output-format", "json+"],
+            ["--numjobs", self.parameters.get_engine_instances_count()],
+            ["--name", name],
+            ["--invalidate", 1],
+            ["--log_avg_msec", self.log_avg_msec],
+        ]
+        for log_type in ["bw", "lat", "hist", "iops"]:
+            enforced_items.append(f"--write_{log_type}_log=fio/{name}_{log_type}.log")
+
+        for enforced_item in enforced_items:
+            args = remove_arg(args, enforced_item)
+            if isinstance(enforced_item, str):
+                args.append(enforced_item)
+            else:
+                args.append(f"{enforced_item[0]}={enforced_item[1]}")
+
+        return args
+
+    def parse_cmd(self, stdout: bytes, stderr: bytes) -> dict[str, Any]:
+        if self.skip:
+            return self.parameters.get_result_format() | self.empty_result()
+        try:
+            ret = json.loads(stdout)
+        except json.decoder.JSONDecodeError:
+            print(
+                f"{self.parameters.get_name_with_position()}: Cannot load fio's JSON output"
```

Comment:
this should print stdout and stderr to debug why fio's JSON output can't be parsed.